Source/EasyNetQ/QueueArgumentsExtensions.cs

```diff
@@ -0,0 +1,145 @@
+    /// </summary>
+    /// <param name="arguments">The configuration instance</param>
+    /// <param name="deadLetterExchange">The dead letter exchange to set</param>
+    /// <returns>The same <paramref name="arguments"/></returns>
+    public static Arguments WithQueueDeadLetterExchange(this Arguments arguments, string deadLetterExchange) =>
+        arguments.WithQueueArgument(QueueArgument.DeadLetterExchange, deadLetterExchange);
+
+    /// <summary>
+    ///     Sets dead letter routing key.
+    ///     If set, will route message when this message is dead-lettered with the routing key specified.
+    ///     If not set, message will be routed with the same routing keys they were originally published with.
+    /// </summary>
+    /// <param name="arguments">The configuration instance</param>
+    /// <param name="deadLetterRoutingKey">The dead letter routing key to set</param>
+    /// <returns>The same <paramref name="arguments"/></returns>
+    public static Arguments WithQueueDeadLetterRoutingKey(this Arguments arguments, string deadLetterRoutingKey) =>
+        arguments.WithQueueArgument(QueueArgument.DeadLetterRoutingKey, deadLetterRoutingKey);
+
+    /// <summary>
+    ///     Sets queue master locator.
+    ///     Valid types are min-masters, client-local and random, see <see cref="QueueMasterLocator"/>.
+    /// </summary>
+    /// <param name="arguments">The configuration instance</param>
+    /// <param name="queueMasterLocator">The queue master locator to set</param>
+    /// <returns>The same <paramref name="arguments"/></returns>
+    public static Arguments WithQueueMasterLocator(this Arguments arguments, string queueMasterLocator = QueueMasterLocator.MinMasters) =>
+        arguments.WithQueueArgument(QueueArgument.QueueMasterLocator, queueMasterLocator);
+
+    /// <summary>
+    ///     Sets dead letter strategy.
+    ///     Valid types are at-least-once and at-most-once, see <see cref="DeadLetterStrategy"/>.
+    /// </summary>
+    /// <param name="arguments">The configuration instance</param>
+    /// <param name="deadLetterStrategy">The dead letter strategy to set</param>
+    /// <returns>The same <paramref name="arguments"/></returns>
+    public static Arguments WithQueueDeadLetterStrategy(this Arguments arguments, string deadLetterStrategy = DeadLetterStrategy.AtMostOnce) =>
+        arguments.WithQueueArgument(QueueArgument.DeadLetterStrategy, deadLetterStrategy);
+
+    /// <summary>
+    ///     Sets overflow type to configure overflow behaviour.
+    ///     Valid types are drop-head, reject-publish and reject-publish-dlx, see <see cref="OverflowType"/>.
+    /// </summary>
+    /// <param name="arguments">The configuration instance</param>
+    /// <param name="overflowType">The overflow type to set</param>
+    /// <returns>The same <paramref name="arguments"/></returns>
+    public static Arguments WithQueueOverflowType(this Arguments arguments, string overflowType = OverflowType.DropHead) =>
```

Comment:
these default parameter values allow to cal method without params and for me this call looks unintelligible

- `= DeadLetterStrategy.AtMostOnce`
- `= OverflowType.DropHead`

Reply:
I have removed all default values here.